src/main/java/ui/Ui.java

```diff
@@ -0,0 +1,48 @@
+package ui;
+
+import exception.DukeException;
+import parser.Parser;
+import storage.Storage;
+import task.TaskList;
+import java.util.Scanner;
+
+/**
+ * Interface that the user interacts with.
+ * Involves a scanner to read user inputs.
+ */
+public class Ui {
+
+    TaskList tasks;
+
+    public Ui(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Starts conversation with user by combining Parser and Storage.
+     * @param parser Instance of Parser to process user commands.
+     * @param storage Instance of storage to update data in storage
+     * @throws DukeException If invalid input message is entered.
+     */
+    public void startConversation(Parser parser, Storage storage) throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String inputString = sc.nextLine();
```

Comment:
Perhaps the naming of this could be "stringInput" to emphasise that it is referring to an input, that is of String type - or simply "input" since we have already established that it is of type String. 👍

Reply:
Hm I see... Thanks!